### PR TITLE
Stop animating in deactivate()

### LINKED ISSUE #1254: TabBarSelection clients need to drop their performance in deactivate
Not dispose, otherwise when we reuse them they'll still be listening even if they're in the wrong part of the tree.

### PR DESCRIPTION
**Pull request by HansMuller**
Fixes https://github.com/flutter/flutter/issues/1254

### CODE REVIEW

**Comment by HansMuller** [comment]
@Hixie

**Comment by Hixie** [comment]
test?

lgtm

File: packages/flutter/lib/src/material/tabs.dart [hunk 1]
```diff
@@ -520,11 +514,13 @@ class TabBarSelectionState<T> extends State<TabBarSelection<T>> {
   }
 
   void deactivate() {
+    _performance.stop();
     for (TabBarSelectionPerformanceListener listener in _performanceListeners.toList()) {
       listener.handleSelectionDeactivate();
       unregisterPerformanceListener(listener);
     }
     assert(_performanceListeners.isEmpty);
+    PageStorage.of(context)?.writeState(context, _value);
```
Comment (Hixie) [inline packages/flutter/lib/src/material/tabs.dart]: this should happen wherever the value is changed

**Comment by Hixie** [comment]
@HansMuller is this still on your radar?

**Pull request closed without merging**